### PR TITLE
feat: added grpc-out option

### LINKED ISSUE #230: Add --grpc_out option
Would be nice to have --grpc_out in namely/protoc-all along with --js_out

### PR DESCRIPTION
**Pull request by wespen**
Added option for grpc-out feature to node file generation.
Enables option to use grpc-js or grpc package or alternatively uses default generate_package_definition for compatibility with both.
Based on @artificial-aidan work for js-out

Closes #230 I opened

### CODE REVIEW

**Comment by ayZagen** [comment]
@nycdotnet @ido-namely sorry to ping, but is there any particular reason why this is not merged? If so, i could help. having this option would be great for us.

**Comment by nycdotnet** [comment]
Taking a look thanks @wespen and @ayZagen

File: all/entrypoint.sh [hunk 1]
```diff
@@ -36,7 +36,8 @@ printUsage() {
                                              (deprecated. Please use --with-openapi-json-names)"
     echo " --with-openapi-json-names      Use with --with-gateway flag. Generated OpenAPI file will use JSON names instead of protobuf names."
     echo " --js-out                       This option overrides the 'js_out=' argument in the grpc-node and grpc-web code generation. Defaults to 'import_style=commonjs'."
-    echo " --grpc-web-out                 This option overrides the 'grpc-web_out=' argument in the grpc-web code generation. Defaults to 'import_style=typescript'."
+    echo " --grpc-out                     This option overrides the 'grpc_out=' argument in the grpc-node and grpc-web code generation. Defaults to 'import_style=commonjs'."
```
Comment (nycdotnet) [inline all/entrypoint.sh:39]:
```suggestion
    echo " --grpc-out                     This option allows overriding the left-half of the 'grpc_out=' argument (before the colon) with grpc-node and grpc-web code generation. Defaults to generate_package_definition"
```

File: all/entrypoint.sh [hunk 2]
```diff
@@ -36,7 +36,8 @@ printUsage() {
...
+    echo " --grpc-web-out                 This option overrides the 'grpc-web_out=' argument in the grpc-web code generation to use grpc_js vs grpc. Defaults to 'generate_package_definition' for compatibility with both."
```
Comment (nycdotnet) [inline all/entrypoint.sh:40]:
```suggestion
    echo " --grpc-web-out                 This option overrides the 'grpc-web_out=' argument in the grpc-web code generation.  Defaults to 'import_style=typescript'."
```
```suggestion
    echo " --grpc-web-out                 This option overrides the 'grpc-web_out=' argument in the grpc-web code generation to use grpc_js vs grpc. Defaults to 'generate_package_definition' for compatibility with both."
```

File: README.md [hunk 3]
```diff
@@ -98,6 +98,8 @@ $ docker run ... namely/protoc-all -f protorepo/catalog/catalog.proto -l go
 
 `--grpc-web-out <string>` to modify the `grpc-web_out=` options for web code generation
 
+`--grpc_out <string>` to modify the `grpc_out=` options for node and web code generation
```
Comment (nycdotnet) [inline README.md:101]:
```suggestion
`--grpc-out <string>` to modify the `grpc_out=` options for node and web code generation.  See https://www.npmjs.com/package/grpc-tools for more details.
```
```suggestion
`--grpc_out <string>` to modify the `grpc_out=` options for node and web code generation
```

**Review by nycdotnet — COMMENTED** [review]
Hi - I think we need the following changes.  Does this make sense?

**Comment by wespen** [comment]
Jup, will be easier to understand and to future-proof it for any new options that might come out.

**Comment by nycdotnet** [comment]
@ido-namely is this satisfactory to you?

**Comment by ido-namely** [comment]
Thanks @wespen and @nycdotnet .
@wespen if you don't mind bumping he version [here](https://github.com/namely/docker-protoc/blob/ce4a5fc31e7cbb367f7c38ad3982c0387c58ade7/variables.sh#L10) and we can get this merged.

**Review by ido-namely — COMMENTED** [review]
we just need to bump the version number here https://github.com/namely/docker-protoc/blob/ce4a5fc31e7cbb367f7c38ad3982c0387c58ade7/variables.sh#L10.
Otherwise LGTM. Thanks!

File: variables.sh [hunk 4]
```diff
@@ -7,7 +7,7 @@ GRPC_JAVA_VERSION=${GRPC_JAVA_VERSION:-1.34}
 GRPC_WEB_VERSION=${GRPC_WEB_VERSION:-1.2.1}
 GRPC_GATEWAY_VERSION=${GRPC_GATEWAY_VERSION:-2.0.1}
 GO_VERSION=${GO_VERSION:-1.14}
-BUILD_VERSION=${BUILD_VERSION:-3}
+BUILD_VERSION=${BUILD_VERSION:-4}
```
Comment (ido-namely) [inline variables.sh:10]: I think you need to pull latest/rebase, the current build version is 4

Comment (wespen) [reply variables.sh]: Done. Bumped to 5

**Review by ido-namely — APPROVED** [review]
Thank you for the contribution!

**Review by nycdotnet — APPROVED** [review]
Thanks.  Sorry for the delay!

**Pull request merged**